src/ServarrAPI/Model/UpdateFileRepository.cs

```diff
@@ -49,8 +54,9 @@ public async Task<List<UpdateFileEntity>> Find(string branch, OperatingSystem os
             {
                 var result = await Query(Builder()
                                          .Where<UpdateFileEntity>(f => f.OperatingSystem == os)
-                                         .Where<UpdateEntity>(u => u.Branch == branch)
-                                         .OrderBy($"update.releasedate DESC LIMIT {count}"))
+                                         .Where<UpdateEntity>(u => u.IntVersion <= maxVersionInt &&
```

Comment:
Maybe we should build the where based on if a max version was provided, no use in querying against a max version if it’s just some high placeholder that doesn’t matter.